test/e2e/serial/tests/workload_placement.go

```diff
@@ -616,6 +619,209 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
 				return ok
 			}).WithTimeout(time.Minute).WithPolling(time.Second*5).Should(BeTrue(), "resources not restored on %q", updatedPod.Spec.NodeName)
 		})
+		It("[test_id:48746][tier2] should modify workload post scheduling while keeping the resource requests available across all NUMA node", func() {
+
+			hostsRequired := 2
+			paddedNodes := padder.GetPaddedNodes()
+			paddedNodesSet := sets.NewString(paddedNodes...)
+
+			err := fxt.Client.List(context.TODO(), &nrtList)
+			Expect(err).ToNot(HaveOccurred())
+
+			// so we can't support ATM zones > 2. HW with zones > 2 is rare anyway, so not to big of a deal now.
+			By(fmt.Sprintf("filtering available nodes with at least %d NUMA zones", 2))
+			nrtCandidates := e2enrt.FilterZoneCountEqual(nrtList.Items, 2)
+			if len(nrtCandidates) < hostsRequired {
+				Skip(fmt.Sprintf("not enough nodes with 2 NUMA Zones: found %d", len(nrtCandidates)))
+			}
+
+			singleNUMAPolicyNrts := e2enrt.FilterByPolicies(nrtList.Items, []nrtv1alpha1.TopologyManagerPolicy{nrtv1alpha1.SingleNUMANodePodLevel, nrtv1alpha1.SingleNUMANodeContainerLevel})
+			nodesNameSet := e2enrt.AccumulateNames(singleNUMAPolicyNrts)
+
+			// the only node which was not padded is the targetedNode
+			// since we know exactly how the test setup looks like we expect only targeted node here
+			targetNodeNameSet := nodesNameSet.Difference(paddedNodesSet)
+			Expect(targetNodeNameSet.Len()).To(Equal(1), "could not find the target node")
+
+			targetNodeName, ok := targetNodeNameSet.PopAny()
+			Expect(ok).To(BeTrue())
+
+			targetNrtInitial, err := e2enrt.FindFromList(nrtList.Items, targetNodeName)
+			Expect(err).NotTo(HaveOccurred())
+
+			var replicas int32 = 2
+			podLabels := map[string]string{
+				"test": "test-dp-two-replicas",
+			}
+
+			// calculate base load on the target node
+			baseload, err := nodes.GetLoad(fxt.K8sClient, targetNodeName)
+			Expect(err).ToNot(HaveOccurred(), "missing node load info for %q", targetNodeName)
+			klog.Infof(fmt.Sprintf("computed base load: %s", baseload))
+
+			// get least available CPU and Memory on each NUMA node while taking baseload into consideration
+			cpus := leastAvailableResourceQtyInAllZone(*targetNrtInitial, baseload, corev1.ResourceCPU)
+			mem := leastAvailableResourceQtyInAllZone(*targetNrtInitial, baseload, corev1.ResourceMemory)
+
+			// We want a container to occupy as much resources from a single NUMA nodes as possible in order to prevent another
+			// container to be allocated resources from the same NUMA node. To determine the value of resources, we use the
+			// resource availablity of a NUMA node that has the least amount of resources out of all the NUMA nodes on that
+			// node and request that in the test-deployment.
+			reqResources := corev1.ResourceList{
+				corev1.ResourceCPU:    cpus,
+				corev1.ResourceMemory: mem,
+			}
+
+			nodeSelector := map[string]string{
+				serialconfig.MultiNUMALabel: "2",
+			}
```

Comment:
where was the target node labeled with this label?

Reply:
the intent of the label (and nodeSelector) is to further restrict the worker node pool to nodes with exactly 2 NUMA cells. We should probably revisit this approach someday in the far future.

Reply:
but we must leave the freedom for the scheduler to decide that, otherwise we are "helping" it with the filtering process

Reply:
is not ideal indeed. This was introduced because we had clusters with workers with mixed NUMA nodes (e.g. 2 NUMA + 1 NUMA in the same pool) and, to a lesser extent, because most of our logic **in the tests** expects 2 NUMA nodes and can't work with 4 NUMA nodes or more.

On the other hand, we're forcing the scheduler to pick a node _among_ all the available nodes with 2 NUMA zones, which isn't too bad - the tests are still meaningful.

We should get rid of all the instances of `nodeSelector` someday in the future and be more restrictive in the test setup, yes.

Reply:
This was exactly the case for the cluster where I tested this and narrowing down to nodes that had 2 NUMAs made sense. Also in the context of this test, we want the two replicas to be placed on two NUMA nodes of the node so having nodes that don't have multiple NUMA nodes did not make much sense.

Reply:
fair enough, thank you for elaborating.

Reply:
right